src/ORM/Association/HasMany.php

```diff
@@ -134,6 +181,62 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Deletes/sets null the related objects according to the dependency between source and targets and foreign key nullability
+     * Skips deleting records present in $remainingEntities
+     *
+     * @param array $properties array of foreignKey properties
+     * @param EntityInterface $entity the entity which should have its associated entities unassigned
+     * @param Table $target The associated table
+     * @param array $remainingEntities Entities that should not be deleted
+     * @return void
+     */
+    protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $remainingEntities)
+    {
+        $primaryKey = (array)$target->primaryKey();
+        $mustBeDependent = (!$this->_foreignKeyAcceptsNull($target, $properties) || $this->dependent());
+        $conditions = [
+            'AND' => [
+                'NOT' => [
+                    'OR' => array_map(
+                        function ($ent) use ($primaryKey) {
+                            return $ent->extract($primaryKey);
+                        },
+                        $remainingEntities
+                    )
+                ],
+                $properties
+            ]
+        ];
+
+        if ($mustBeDependent) {
+                $target->deleteAll($conditions);
```

Comment:
@lorenzo Yes, I got that, but I still didn't code it yet =P
I wil provide it ASAP